File: forlab/ForLab.Data/DbModels/ForecastingSchema/ForecastResult.cs
using ForLab.Data.BaseModeling;
using ForLab.Data.DbModels.LookupSchema;
using ForLab.Data.DbModels.ProductSchema;
using ForLab.Data.DbModels.TestingSchema;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace ForLab.Data.DbModels.ForecastingSchema
{
    [Table("ForecastResults", Schema = "Forecasting")]
    public class ForecastResult : BaseEntity
    {
        public int ForecastInfoId { get; set; }
        public int LaboratoryId { get; set; }
        public int? TestId { get; set; }
        public int ProductId { get; set; }
        public decimal AmountForecasted { get; set; } // which will be fetched from the machine learning model
        public decimal TotalValue { get; set; }
        public DateTime DurationDateTime { get; set; }
        public string Period { get; set; }
        public decimal PackSize { get; set; }
        public int PackQty { get; set; }
        public decimal PackPrice { get; set; }
        public decimal TotalPrice { get; set; }
        public int ProductTypeId { get; set; }

        public virtual ForecastInfo ForecastInfo { get; set; }
        public virtual Test Test { get; set; }
        public virtual Product Product { get; set; }
        public virtual Laboratory Laboratory { get; set; }
        public virtual ProductType ProductType { get; set; }
    }
}

File: forlab/ForLab.Web/ClientApp/src/@core/models/forecasting/ForecastResult.ts
import { BaseEntityDto } from '../common/BaseEntityDto';
import { BaseFilter } from '../common/BaseFilter';

export class ForecastResultDto extends BaseEntityDto {
  forecastInfoId: number = null;
  laboratoryId: number = null;
  testId: number = null;
  productId: number = null;
  amountForecasted: number = null;
  totalValue: number = null;
  durationDateTime: number = null;
  period: number = null;
  packSize: number = null;
  packQty: number = null;
  packPrice: number = null;
  totalPrice: number = null;
  productTypeId: number = null;
  
  // UI
  forecastInfoName: string = null;
  laboratoryName: string = null;
  testName: string = null;
  productName: string = null;
  productTypeName: string = null;
}

export class ForecastResultFilterDto extends BaseFilter {
  forecastInfoId: number = null;
  laboratoryId: number = null;
  testId: number = null;
  productId: number = null;
  productTypeId: number = null;
}